Source/DafnyDriver/TextLogger.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Boogie;
+
+namespace Microsoft.Dafny;
+
+public class TextLogger {
+  private TextWriter tw;
+
+  public void Initialize(Dictionary<string, string> parameters) {
+    tw = parameters.TryGetValue("LogFileName", out string filename) ? new StreamWriter(filename) : Console.Out;
+  }
+
+  public void LogResults(List<(Implementation, VerificationResult)> verificationResults) {
```

Comment:
Would you mind attaching some sample output, just to double check that the formatting makes sense etc.? The CHECK testing is perfect but omits some of that since it has to regex-away details.

Reply:
Here's without `/vcsSplitOnEveryAssert`:

````
Results for Impl$$_module.__default.M
  Overall outcome: Errors
  Overall time: 00:00:00.4675430
  Overall resource count: 10947

  Assertion batch 0:
    Outcome: Invalid
    Duration: 00:00:00.1252410
    Resource count: 10947

    Assertions:
      /Users/aarotomb/Repositories/dafny/Test/logger/TextLogger.dfy(18,14): divisor is always non-zero.
      /Users/aarotomb/Repositories/dafny/Test/logger/TextLogger.dfy(19,9): assertion always holds
````

And with:
````
Results for Impl$$_module.__default.M
  Overall outcome: Errors
  Overall time: 00:00:00.5055420
  Overall resource count: 22035

  Assertion batch 1:
    Outcome: Valid
    Duration: 00:00:00.0892450
    Resource count: 3267

    Assertions:

  Assertion batch 2:
    Outcome: Valid
    Duration: 00:00:00.0343600
    Resource count: 8473

    Assertions:
      /Users/aarotomb/Repositories/dafny/Test/logger/TextLogger.dfy(18,14): divisor is always non-zero.

  Assertion batch 3:
    Outcome: Invalid
    Duration: 00:00:00.0191820
    Resource count: 10295

    Assertions:
      /Users/aarotomb/Repositories/dafny/Test/logger/TextLogger.dfy(19,9): assertion always holds
````

Reply:
The names are still in Boogie format. Improving that is something I'd still like to do soon, but will require a bit of work.

Reply:
Thanks, LGTM - I mostly wanted to double check the formatting on the durations was reasonable.

Interesting how a batch with zero assertions still has a non-zero resource count isn't it! I assume that has to be assertions in the `DafnyPrelude.bpl` or some other kind of fixed cost.

Reply:
That's something I've been meaning to investigate for a while. I looked at one instance and it was just `true` (with some background assumptions, which is probably what makes it take non-zero resources). I wonder if it's an artifact of splitting. Do you know, @keyboardDrummer?